From pearl-miui-camera-6.6-compat
Subject: [PATCH] Fix crash when obfu_res drawable is missing

The encrypted drawable loader tries to open assets under obfu_res/.
Some whitelisted drawables (e.g. ic_west_coast_princess) do not have a
corresponding obfu_res asset in this build, causing FileNotFoundException
and a main-interface crash when the top menu arrow is tapped.

Catch FileNotFoundException in If.a.c and fall back to the normal
Resources.getDrawable(resourceId) path so the camera no longer crashes.

---
 smali_classes3/If/a.smali | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/smali_classes3/If/a.smali b/smali_classes3/If/a.smali
--- a/smali_classes3/If/a.smali
+++ b/smali_classes3/If/a.smali
@@ -229,6 +229,8 @@
 
     invoke-virtual {p0, p1, p2, v0, v3}, Landroid/content/res/Resources;->getValueForDensity(IILandroid/util/TypedValue;Z)V
 
+    move v4, p1
+
     iget-object p1, v0, Landroid/util/TypedValue;->string:Ljava/lang/CharSequence;
 
     invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
@@ -245,9 +247,23 @@
 
     move-result-object p2
 
+    :try_start_obfu
     invoke-virtual {p2, p1}, Landroid/content/res/AssetManager;->open(Ljava/lang/String;)Ljava/io/InputStream;
 
     move-result-object p2
+    :try_end_obfu
+    .catch Ljava/io/FileNotFoundException; {:try_start_obfu .. :try_end_obfu} :catch_obfu
+
+    goto :try_start_0
+
+    :catch_obfu
+    move-exception v0
+
+    invoke-virtual {p0, v4}, Landroid/content/res/Resources;->getDrawable(I)Landroid/graphics/drawable/Drawable;
+
+    move-result-object p0
+
+    return-object p0
 
     :try_start_0
     invoke-static {p2}, Lfv/l;->e(Ljava/lang/Object;)V
